File: zig/eliuds-eggs/eliuds_eggs.zig
const std = @import("std");
pub fn eggCount(number: usize) usize {
    var num: usize = number;
    var result: usize = 0;
    while (num != 0) : (num /= 2) {
        if (num % 2 == 1) {
            result += 1;
        }
    }
    std.debug.print("{d}", .{num});
    return result;
}

File: lua/eliuds-eggs/eliuds-eggs.lua
local EliudsEggs = {}

function EliudsEggs.egg_count(number)
  local results = 0
  while number > 0 do
    results = results + (number & 1)
    number = number >> 1
  end
  --account for missing 1
  return results
end

return EliudsEggs
